src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Parser.cs

```diff
@@ -0,0 +1,286 @@
+            {
+                return null;
+            }
+
+            DiagnosticDescriptor? errorDescriptor = null;
+            RegexMethod? regexMethod = null;
+            foreach (AttributeData attributeData in boundAttributes)
+            {
+                // If we already encountered an error, stop looking at this method's attributes.
+                if (errorDescriptor is not null)
+                {
+                    break;
+                }
+
+                // If this isn't 
+                if (!attributeData.AttributeClass.Equals(regexGeneratorAttributeSymbol))
+                {
+                    continue;
+                }
+
+                if (attributeData.ConstructorArguments.Any(ca => ca.Kind == TypedConstantKind.Error))
+                {
+                    errorDescriptor = DiagnosticDescriptors.InvalidRegexGeneratorAttribute;
+                    break;
+                }
+
+                ImmutableArray<TypedConstant> items = attributeData.ConstructorArguments;
+                if (items.Length is > 0 and <= 3 && items[0].Value is string pattern)
+                {
+                    switch (items.Length)
+                    {
+                        case 1:
+                            regexMethod = new RegexMethod { Pattern = pattern };
+                            break;
+
+                        case 2:
+                            regexMethod = new RegexMethod { Pattern = pattern, Options = items[1].Value as int?, };
+                            break;
+
+                        case 3:
+                            regexMethod = new RegexMethod { Pattern = pattern, Options = items[1].Value as int?, MatchTimeout = items[2].Value as int?, };
+                            break;
+                    }
+                }
+                else
+                {
+                    errorDescriptor = DiagnosticDescriptors.InvalidRegexGeneratorAttribute;
+                }
+            }
+
+            if (errorDescriptor is not null)
+            {
+                return Diagnostic.Create(errorDescriptor, methodSyntax.GetLocation());
+            }
+
+            if (regexMethod is null)
+            {
+                return null;
+            }
+
+            if (regexMethod.Pattern is null)
+            {
+                return Diagnostic.Create(DiagnosticDescriptors.InvalidRegexArguments, methodSyntax.GetLocation(), "(null)");
+            }
+
+            if (!regexMethodSymbol.IsPartialDefinition ||
+                !regexMethodSymbol.IsStatic ||
+                regexMethodSymbol.Parameters.Length != 0 ||
+                regexMethodSymbol.Arity != 0 ||
+                !regexMethodSymbol.ReturnType.Equals(regexSymbol))
+            {
+                return Diagnostic.Create(DiagnosticDescriptors.RegexMethodMustHaveValidSignature, methodSyntax.GetLocation());
+            }
+
+            if (typeDec.SyntaxTree.Options is CSharpParseOptions { LanguageVersion: < LanguageVersion.CSharp10 })
```

Comment:
Is this because our code gen uses C# 10 features?

Reply:
I think technically I ended up using only C# 9, but I've not gone back to reconfirm.